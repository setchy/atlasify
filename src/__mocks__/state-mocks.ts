import {
  type Account,
  type AtlassifyState,
  type AtlassifyUser,
  type AuthState,
  type Link,
  OpenPreference,
  type SettingsState,
  Theme,
  type Token,
  type Username,
} from '../types';

export const mockAtlassifyUser: AtlassifyUser = {
  login: 'atlas' as Username,
  name: 'Atlassian',
  id: '123456789',
  avatar: 'https://avatars.githubusercontent.com/u/583231?v=4' as Link,
};

export const mockAtlassianCloudAccount: Account = {
  platform: 'Atlassian Cloud',
  method: 'API Token',
  token: 'token-123-456' as Token,
  user: mockAtlassifyUser,
};

export const mockAuth: AuthState = {
  accounts: [mockAtlassianCloudAccount],
};

const mockAppearanceSettings = {
  theme: Theme.LIGHT,
  zoomPercentage: 100,
};

const mockNotificationSettings = {
  markAsReadOnOpen: true,
  delayNotificationState: false,
  fetchOnlyUnreadNotifications: true,
  groupNotificationsByProduct: false,
};

const mockSystemSettings = {
  openLinks: OpenPreference.FOREGROUND,
  keyboardShortcutEnabled: true,
  showNotificationsCountInTray: true,
  showSystemNotifications: true,
  playSoundNewNotifications: true,
  useAlternateIdleIcon: false,
  openAtStartup: false,
};

const mockFilters = {
  filterCategories: [],
  filterReadStates: [],
  filterProducts: [],
};

export const mockSettings: SettingsState = {
  ...mockAppearanceSettings,
  ...mockNotificationSettings,
  ...mockSystemSettings,
  ...mockFilters,
};

export const mockState: AtlassifyState = {
  auth: mockAuth,
  settings: mockSettings,
};
