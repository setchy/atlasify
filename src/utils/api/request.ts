import axios, { type Method, type AxiosPromise } from 'axios';
import type { Account, Token, Username } from '../../types';
import { Constants } from '../constants';

export function performPostRequest(
  account: Account,
  data = {},
): AxiosPromise | null {
  const auth = btoa(`${account.user.login}:${account.token}`);

  const method: Method = 'POST';

  return performApiRequest(auth, method, data);
}

export function performHeadRequest(
  username: Username,
  token: Token,
): AxiosPromise | null {
  const auth = btoa(`${username}:${token}`);

  const method: Method = 'HEAD';

  return performApiRequest(auth, method, {});
}

function performApiRequest(
  auth: string,
  method: Method,
  data = {},
): AxiosPromise | null {
  const url = Constants.ATLASSIAN_URLS.API;

  axios.defaults.headers.common.Accept = '*/*';
  axios.defaults.headers.common.Authorization = `Basic ${auth}`;
  axios.defaults.headers.common['Cache-Control'] = 'no-cache';
  axios.defaults.headers.common['Content-Type'] = 'application/json';

  return axios({ method, url, data });
}
