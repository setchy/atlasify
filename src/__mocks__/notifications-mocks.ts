import { BitbucketIcon, ConfluenceIcon } from '@atlaskit/logo';
import type {
  AccountNotifications,
  AtlassifyNotification,
  Link,
} from '../types';
import { mockAtlassianCloudAccount } from './state-mocks';

export const mockAtlassifyNotifications: AtlassifyNotification[] = [
  {
    account: mockAtlassianCloudAccount,
    id: '138661096',
    title: 'I am a robot and this is a test!',
    updated_at: '2020-09-13T21:12:16.662Z',
    entity: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    path: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    product: {
      name: 'bitbucket',
      icon: BitbucketIcon,
    },
    actor: {
      displayName: 'atlassify-app',
      avatarURL:
        'https://avatars.githubusercontent.com/u/133795385?s=200&v=4' as Link,
    },
    category: 'direct',
    readState: 'unread',
    url: 'https://api.github.com/notifications/threads/138661096' as Link,
  },
  {
    id: '148827438',
    title: 'I am a robot and this is a test!',
    updated_at: '2020-09-13T21:12:16.662Z',
    entity: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    path: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    product: {
      name: 'confluence',
      icon: ConfluenceIcon,
    },
    actor: {
      displayName: 'atlassify-app',
      avatarURL:
        'https://avatars.githubusercontent.com/u/133795385?s=200&v=4' as Link,
    },
    category: 'direct',
    readState: 'unread',
    account: mockAtlassianCloudAccount,
    url: 'https://api.github.com/notifications/threads/148827438' as Link,
  },
];

export const mockSingleNotification: AtlassifyNotification =
  mockAtlassifyNotifications[0];

export const mockAccountNotifications: AccountNotifications[] = [
  {
    account: mockAtlassianCloudAccount,
    notifications: mockAtlassifyNotifications,
    hasNextPage: false,
    error: null,
  },
];

export const mockAccountNotificationsWithMorePages: AccountNotifications[] = [
  {
    account: mockAtlassianCloudAccount,
    notifications: mockAtlassifyNotifications,
    hasNextPage: true,
    error: null,
  },
];

export const mockSingleAccountNotifications: AccountNotifications[] = [
  {
    account: mockAtlassianCloudAccount,
    notifications: [mockAtlassifyNotifications[0]],
    hasNextPage: false,
    error: null,
  },
];
