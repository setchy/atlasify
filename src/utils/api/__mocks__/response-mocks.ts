import { BitbucketIcon } from '@atlaskit/logo';
import { mockAtlassianCloudAccount } from '../../../__mocks__/state-mocks';
import type { AtlassifyNotification, Link } from '../../../types';

export const mockAtlassifyNotification: AtlassifyNotification[] = [
  {
    account: mockAtlassianCloudAccount,
    id: '138661096',
    title: 'I am a robot and this is a test!',
    unread: true,
    updated_at: '2024-09-13T21:12:16.662Z',
    entity: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    path: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    product: {
      name: 'bitbucket',
      icon: BitbucketIcon,
    },
    actor: {
      displayName: 'atlassify-app',
      avatarURL:
        'https://avatars.githubusercontent.com/u/133795385?s=200&v=4' as Link,
    },
    category: 'direct',
    readState: 'unread',
    url: 'https://api.github.com/notifications/threads/138661096' as Link,
  },
  {
    id: '148827438',
    title: 'I am a robot and this is a test!',
    unread: true,
    updated_at: '2024-09-13T21:12:16.662Z',
    entity: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    path: {
      title: 'Repository',
      iconUrl: 'https://github.atlassify.io/favicon.ico' as Link,
      url: 'https://github.atlassify.io/myorg/notifications-test' as Link,
    },
    product: {
      name: 'bitbucket',
      icon: BitbucketIcon,
    },
    actor: {
      displayName: 'atlassify-app',
      avatarURL:
        'https://avatars.githubusercontent.com/u/133795385?s=200&v=4' as Link,
    },
    category: 'direct',
    readState: 'unread',
    account: mockAtlassianCloudAccount,
    url: 'https://api.github.com/notifications/threads/148827438' as Link,
  },
];

export const mockSingleNotification: AtlassifyNotification =
  mockAtlassifyNotification[0];
