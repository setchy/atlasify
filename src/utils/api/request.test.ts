import axios from 'axios';
import { mockAtlassianCloudAccount } from '../../__mocks__/state-mocks';
import type { Link } from '../../types';
import { performPostRequest } from './request';

jest.mock('axios');

const url = 'https://team.atlassian.net/gateway/api/graphql' as Link;
const method = 'POST';

describe('utils/api/request.ts', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should make a request with the correct parameters', async () => {
    const data = { key: 'value' };

    await performPostRequest(mockAtlassianCloudAccount, data);

    expect(axios).toHaveBeenCalledWith({
      method,
      url,
      data,
    });

    expect(axios.defaults.headers.common).toMatchSnapshot();
  });

  it('should make a request with the correct parameters and default data', async () => {
    const data = {};
    await performPostRequest(mockAtlassianCloudAccount, data);

    expect(axios).toHaveBeenCalledWith({
      method,
      url,
      data,
    });

    expect(axios.defaults.headers.common).toMatchSnapshot();
  });
});
