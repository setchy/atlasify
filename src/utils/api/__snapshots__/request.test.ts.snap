// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/api/request.ts should make a head request with the correct parameters 1`] = `
{
  "Accept": "*/*",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;

exports[`utils/api/request.ts should make a post request with the correct parameters 1`] = `
{
  "Accept": "*/*",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;
